source/MaterialXContrib/MaterialXNode/MaterialXData.cpp

```diff
@@ -1,49 +1,93 @@
-#include <MaterialXData.h>
-#include <Util.h>
+#include "MaterialXData.h"
+#include "MaterialXUtil.h"
 
 #include <MaterialXFormat/XmlIo.h>
 #include <MaterialXGenOgsXml/GlslFragmentGenerator.h>
 #include <MaterialXGenShader/Util.h>
 
-MaterialXData::MaterialXData(const std::string& materialXDocumentPath, 
-                             const std::string& elementPath, 
-                             const MaterialX::FileSearchPath& librarySearchPath)
-    : _shaderGenerator(mx::GlslFragmentGenerator::create()),
-      _genContext(_shaderGenerator),
-      _librarySearchPath(librarySearchPath)
+namespace
 {
-    setRenderableElement(materialXDocumentPath, elementPath);
+mx::DocumentPtr createDocument( const std::string& materialXDocumentPath,
```

Comment:
Making this a free function because it doesn't depend on the state of MaterialXData.